pallets/nfts/src/features/approvals.rs

```diff
@@ -165,10 +179,197 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	/// of items in the collection. Optionally, a `deadline` can be specified to set a time limit
+	/// for the approval. The `deadline` is expressed in block numbers and is added to the current
+	/// block number to determine the absolute deadline for the approval. After approving the
+	/// transfer, the function emits the `TransferApproved` event.
+	///
+	/// This function reserves the required deposit from the `origin` account. If an approval
+	/// already exists, the new amount is added to such existing approval.
+	///
+	/// - `origin`: The account grants permission to approve the transfer.
+	/// - `collection`: The identifier of the collection.
+	/// - `delegate`: The account that will be allowed to take control of items in the collection
+	///   that owned by the `origin`.
+	/// - `maybe_deadline`: The optional deadline (in block numbers) specifying the time limit for
+	///   the approval.
+	pub(crate) fn do_approve_collection_transfer(
+		origin: T::AccountId,
+		collection: T::CollectionId,
+		delegate: T::AccountId,
+		maybe_deadline: Option<BlockNumberFor<T>>,
+	) -> DispatchResult {
+		ensure!(
+			Self::is_pallet_feature_enabled(PalletFeature::Approvals),
+			Error::<T, I>::MethodDisabled
+		);
+
+		let collection_config = Self::get_collection_config(&collection)?;
+		ensure!(
+			collection_config.is_setting_enabled(CollectionSetting::TransferableItems),
+			Error::<T, I>::ItemsNonTransferable
+		);
+		let now = frame_system::Pallet::<T>::block_number();
+		let deadline = maybe_deadline.map(|d| d.saturating_add(now));
+
+		CollectionApprovals::<T, I>::try_mutate_exists(
+			(&collection, &origin, &delegate),
+			|maybe_approval| -> DispatchResult {
+				let deposit_required = T::CollectionApprovalDeposit::get();
+				let mut current_deposit = match maybe_approval.take() {
+					Some((_, deposit)) => deposit,
+					None => Zero::zero(),
+				};
+
+				if current_deposit < deposit_required {
+					T::Currency::reserve(&origin, deposit_required - current_deposit)?;
+					current_deposit = deposit_required;
+				}
+				*maybe_approval = Some((deadline, current_deposit));
+				Ok(())
+			},
+		)?;
+
+		Self::deposit_event(Event::TransferApproved {
+			collection,
+			item: None,
+			owner: origin,
+			delegate,
+			deadline,
+		});
+
+		Ok(())
+	}
+
+	/// Cancels the transfer of items in the collection that owned by the origin to a delegate.
+	///
+	/// This function is used to cancel the approval for the transfer of items in the `collection`
+	/// that owned by the `origin` to a `delegate`. After canceling the approval, the function
+	/// returns the `origin` back the deposited fund and emits the `ApprovalCancelled` event.
+	///
+	/// - `origin`: The account grants permission to cancel the transfer.
+	/// - `collection`: The identifier of the collection.
+	/// - `delegate`: The account that was previously allowed to take control of items in the
+	///   collection that owned by the origin.
```

Comment:
that owned is not proper english, this is the case in many places in the code, please change accordingly (e.g. "... collection owned by"*)